[ZEPPELIN-6242] Refactor initAuth in Neo4jConnectionManager for safer enum parsing and default handling

eunhwa99 opened the pull request:
### What is this PR for?
This PR improves the Neo4j authentication initialization logic by providing a default authentication type of `NONE` when the `neo4j.auth.type` property is absent.  
It enhances enum parsing with explicit try-catch handling to deliver clear, user-friendly error messages for unsupported or invalid authentication types.  

Additionally, a defensive default branch is not removed in the switch statement to satisfy the compiler and guard against unforeseen enum values.

### What type of PR is it?
Refactoring

### Todos
* [x] Provide default `NONE` value when `neo4j.auth.type` is missing  
* [x] Wrap enum parsing in try-catch with explicit error messages  
* [X] Add unit test for the modified code.

### What is the Jira issue?
* [ZEPPELIN-6242](https://issues.apache.org/jira/browse/ZEPPELIN-6242)

### How should this be tested?
* Add new unit test for the modified code.
   * Tests for missing auth type defaulting to NONE, invalid auth types throwing clear exceptions, and successful BASIC/NONE authentications should be confirmed

### Screenshots (if appropriate)
N/A

### Questions:
* Does the license files need to update? No  
* Is there breaking changes for older versions? No  
* Does this needs documentation? No

Reamer commented on neo4j/src/main/java/org/apache/zeppelin/graph/neo4j/Neo4jConnectionManager.java:
```diff
@@ -75,7 +74,21 @@ public class Neo4jConnectionManager {
+  public enum Neo4jAuthType {
+    NONE, BASIC;
+
+    public static Neo4jAuthType fromString(String value) {
```
This method is unnecessary. The `toUpperCase()` should still be in the code.

Reamer commented on neo4j/src/main/java/org/apache/zeppelin/graph/neo4j/Neo4jConnectionManager.java:
```diff
@@ -85,8 +98,9 @@ public Neo4jConnectionManager(Properties properties) {
   }
 
   private AuthToken initAuth(Properties properties) {
-    String authType = properties.getProperty(NEO4J_AUTH_TYPE);
```
Here you can call `toUpperCase()` and specify a default value.

Reamer reviewed (changes requested):
I don't like the extension of the enum. The errors should be caught in a different way.

eunhwa99 commented:
> Sorting the imports and changing the indentation are unnecessary. In addition, the checkstyle test fails because the Neo4jConnectionManagerTest.java class does not contain a new line at the end of the file.

Oh I see. Thanks for the feedback. I reverted the unrelated import and formatting changes and added the missing newline to fix the Checkstyle issue.  8873fb5

Reamer commented on neo4j/src/test/java/org/apache/zeppelin/graph/neo4j/Neo4jConnectionManagerTest.java:
```diff
@@ -0,0 +1,78 @@
+
+    assertTrue(exception.getMessage().contains("Username can't be null"));
+  }
+}
```
It seems that there is no line break at the end of this file. Let's see what the CI says.

eunhwa99 replied:
I added a new line ! Thank you 😄 [5cd2166](https://github.com/apache/zeppelin/pull/4975/commits/5cd216640972a6238a0196f8456f7595d91637e1)

ParkGyeongTae reviewed (commented):
Just a small comment regarding the missing license header in one of the files. Other than that, everything looks good. Thanks for the contribution!

ParkGyeongTae reviewed (approved):
👍 LGTM

Reamer reviewed (approved):
LGTM

Reamer commented:
Merged into master and branch-0.12
